### PR TITLE
Restructure nav bar and add GitHub.

### PR DESCRIPTION
**Pull request by csadorf**
I followed HOOMD-blue's example and also added a link to Github.

### CODE REVIEW

**Comment by csadorf** [comment]
The only annoying part is that the GitHub link is going to a specific repository instead of an organization. Not sure how to remedy that.

File: docs/source/index.rst [hunk 1]
```diff
@@ -14,24 +14,31 @@ It provides a comprehensive overview on what you can do with packages that are p
...
 
    intro
    installation
    quickstart
    tutorial
    reference
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Extra
```
Comment (bdice) [inline docs/source/index.rst]: I dislike the name "Extra." I'm not sure if this categorization makes sense. Maybe "API and Usage"?

File: docs/source/index.rst [hunk 2]
```diff
@@ -14,24 +14,31 @@ It provides a comprehensive overview on what you can do with packages that are p
 If you are new to **signac**, the best place to start is to read the :ref:`introduction` and the :ref:`tutorial`.
 
 
-Contents
---------
-
 .. toctree::
    :maxdepth: 2
+   :caption: Getting started
```
Comment (bdice) [inline docs/source/index.rst]: I'd capitalize the word "Started."

File: docs/source/index.rst (hunk 1 above)
Comment (csadorf) [reply docs/source/index.rst]: I've changed it to "User Guide".

**Comment by csadorf** [comment]
@bdice Addressed your comments.

**Pull request merged**